extension/src/popup/ducks/transactionSubmission.ts

```diff
@@ -244,9 +245,27 @@ export const signWithHardwareWallet = createAsyncThunk<
       bipPath,
       walletType,
       isHashSigningEnabled,
+      isSignSorobanAuthorization,
     },
     thunkApi,
   ) => {
+    if (isSignSorobanAuthorization) {
+      try {
+        const auth = Buffer.from(transactionXDR, "base64");
+
+        const signature = await hardwareSignAuth[walletType]({
+          bipPath,
+          auth,
+          isHashSigningEnabled,
+        });
+
+        return signature;
```

Comment:
returning a signed auth (buffer) is different than returning a signed xdr (string), but combining these allows us to reuse the UX in the view component that handles Ledger error messages/failures